src/components/chat/tool-icon-component.tsx

```diff
@@ -0,0 +1,26 @@
+import React, { useContext } from "react";
+import { useTypeOfTileInDocumentOrCurriculum } from "../../hooks/use-stores";
+import { AppConfigContext } from "../../app-config-context";
+
+interface IProps {
+  documentKey?: string;
+  tileId?: string;
+}
+
+export const ToolIconComponent: React.FC<IProps> = ({documentKey, tileId}) => {
+
+  const tileType = useTypeOfTileInDocumentOrCurriculum(documentKey, tileId);
+  const { appIcons } = useContext(AppConfigContext);
+  
+  if (!documentKey || !tileId || !appIcons ) {
+    return null;
+  }
+
+  if (!tileType) {
+    return <div>{appIcons?.["icon-open-workspace"]}</div>;
+  }
+
+  const iconName = `icon-${tileType.toLowerCase()}-tool`;
+  const TheIcon = appIcons?.[iconName];
+  return <div><TheIcon/></div>;
```

Comment:
Is this container <div> needed or can it just be `return <TheIcon/>;`?

Reply:
I think the compiler was getting confused on the return type of <TheIcon/>, which was why I wrapped it in that <div> element; it is not needed otherwise.

Reply:
It's working locally on my end, but maybe it failed some integration test?

Reply:
This code and the calling code can be simplified by replacing lines 15-25 with:
```suggestion
  const iconName = tileType ? `icon-${tileType.toLowerCase()}-tool` : undefined;
  const Icon = iconName ? appIcons?.[iconName] : undefined;
  return Icon ? <Icon/> : <DocumentIcon/>;
```
utilizing the new document icon I just added.